docs/enhancements/support-node-labeling.md

```diff
@@ -0,0 +1,128 @@
+---
+title: node labeling support
+authors:
+- "@oamizur"
+  creation-date: 2022-12-20
+  last-updated: 2022-12-20
+---
+
+# Support node labeling during installation
+
+## Summary
+
+There was a requirement to support node roles during installation.  These roles 
+are extra roles, which are used by the user and not by any Openshift components.
+These labels should be applied on agent CR which represents a node in a spoke cluster.
+
+## Motivation
+
+Currently, nodes can be assigned with a single role which can be one of two values [ master, worker].
+The may want to add additional roles.  These roles are for the user usage only.
+
+### Goals
+- For ZTP flow: add capability to add and update labels on BMH which will be transferred 
+to agent CR.
+- For boot it yourself flow: add capability to add and update labels to agent CR after agent
+  has been created.
+
+### Non goals
+- Single way to set node labels before agent creation both for ZTP flow, and boot it yourself flow.
+- REST API support.  Node labeling will not be supported under REST API.
```

Comment:
It's already supported, but only in day1 https://github.com/openshift/assisted-service/blob/2b0f66c4cdcdd2dc8556d7f65b7f3e0eac0c958a/swagger.yaml#L4116-L4119

Reply:
OK, so it is non goal.